state.go

```diff
@@ -129,8 +137,19 @@ type StateRunner interface {
 	// Obtain runner's state
 	GetState() State
 	// TODO generic state run action
+	RunState(ctx *StateContext) (State, bool)
```

Comment:
So maybe in that case `Handle()` should be `handle()` and we can use it as a local non-exported helper function?